### PR TITLE
feat(try): add try construct

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #38** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/38" target="_blank">Graphite</a> 
    * **PR #41** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/41" target="_blank">Graphite</a> 
      * **PR #42** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/42" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/axioms/42?utm_source=stack-comment).

**Review by loredanamoanga — APPROVED** [review]

File: src/guard/is-try/is-try.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import type { Try } from '../../type/try'
+
+export function isTry<T extends Try<unknown>>(x: T | unknown): x is T {
+    return x !== null && x !== undefined && typeof x === 'object' && ('success' in x || 'failure' in x)
```
Comment (MickVanDuijn) [inline src/guard/is-try/is-try.ts:4]:
```suggestion
    return true
```
everything is a try in the new definition I guess :laughing:

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/axioms/42?utm_source=stack-comment).


Job Reference: snArrRseLhXioPIul3WC

**Pull request merged**